tests/system/tsigexitcode.nim

```diff
@@ -0,0 +1,20 @@
+discard """
+  joinable: false
+  disabled: windows
+"""
+
+import os, osproc, posix, strutils
+
+proc main() =
+  if paramCount() > 0:
+    let signal = cint parseInt paramStr(1)
+    discard posix.raise(signal)
+  else:
+    # synchronize this list with lib/system/except.nim:registerSignalHandler()
+    let fatalSigs = [SIGINT, SIGSEGV, SIGABRT, SIGFPE, SIGILL, SIGBUS,
+                     SIGPIPE]
+    for s in fatalSigs:
+      let (_, exitCode) = execCmdEx(quoteShellCommand [getAppFilename(), $s])
+      doAssert exitCode == 128 + s, "mismatched exit code for signal " & $s
```

Comment:
```suggestion
      doAssert exitCode == exitStatusLikeShell(s), "mismatched exit code for signal " & $s
```

?

Reply:
Nope, `exitStatusLikeShell()` should only be used for status codes returned via either `system()` or `waitpid()` as it makes use of POSIX macros that are only defined for those uses.